File: 13. Roman to Integer/13. Roman to Integer.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    const roman_map = {
        "I" : 1,  "V" : 5,   "X" : 10,
        "L" : 50, "C" : 100, "D" : 500,
        "M" : 1000
    }

    let total = 0
    let prev = 0
    let curr = 0

    s = s.split("").reverse().join("")

    for (let i = 0; i < s.length; i++){
        curr = roman_map[s.at(i)]

        if (curr < prev) total = total - curr
        else total = total + curr

        prev = curr
    }

    return total
};

File: 13. Roman to Integer/13. Roman to Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_map = {
            'I' : 1,  "V" : 5,  "X" : 10,
            "L" : 50, "C" : 100,"D" : 500, 
            "M" : 1000
        }

        total = 0
        prev = 0

        for c in reversed(s):
            curr = roman_map[c]
            if curr < prev:
                total = total - curr
            else : 
                total = total + curr
            
            prev = curr
        
        return total
                
